### PR TITLE
Add docs for recipes

### PR DESCRIPTION
**Pull request by aem**
I created a new section for templates in the sidebar because I think my next work will be custom templates + ejecting templates, so we'll have more to add there soon. Otherwise these are pretty straightforward

### CODE REVIEW

File: docs/using-recipes.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: Using Recipes
+sidebar_label: Using Recipes
+---
+
```
Comment (flybayer) [inline docs/using-recipes.mdx:5]: Can we add a one sentence value proposition here?

Something like this:

```
Blitz Recipes allow you to install an arbitrary amount of code with one command!
```

And also some bullet points:

```
- ✅ One command: `blitz install [Recipe name]`
- ✅ Many [official recipes](https://github.com/blitz-js/blitz/tree/canary/recipes)
- ✅ Write your own [custom recipes](writing-recipes)
```

File: docs/using-recipes.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Blitz Recipes are predefined formulas for adding npm dependencies or other third-party code to your application. We've taken inspiration from [Gatsby Recipes](https://www.gatsbyjs.org/docs/recipes/) (and are very appreciative of their help in the ideation phase for our own implementation) to improve on the existing flow of adding third-party dependencies to an application.
+
+Rather than following a step-by-step guide from a documentation site that involves manually installing dependencies, updating configuration files, and copying code from example files - a process that can be very error-prone and tedious - we've built the Blitz installer to take a recipe and turn it into a wizard that guides you through the installation, almost exactly like installing software onto your desktop.
```
Comment (flybayer) [inline docs/using-recipes.mdx:8]: This is quite hard to parse.

How about replacing this section with something simpler like this:

```
If you are familiar with the official Next.js examples, Recipes are our replacement for that. To integrate a third-party library, run one command instead of having to look up an example integration.
```

File: docs/using-recipes.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+![Screenshot of the output from the Blitz Recipe for Tailwind CSS. It has added the necessary dependencies, created stylesheets for the developer, and imported them into the app.](/img/tailwind-recipe.png)
```
Comment (flybayer) [inline docs/using-recipes.mdx:10]: What about taking the contents of this screenshot and inlining it into the page? That way it's copiable, will always match our theme, and is better for screen readers.

File: docs/using-recipes.mdx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+### How To Install a Recipe
+
+The full list of Blitz Recipes can be found in the [GitHub repository](https://github.com/blitz-js/blitz/tree/canary/recipes). The name of the recipe is the directory that contains the code. Take the name of the recipe you'd like to install, pass it to [`blitz install`](cli-build), and the installer will guide you through the process, providing you with an explanation for the changes along the way. For example, in the recipes directory today you'll see a recipe for `tailwind`, which installs [Tailwind CSS](https://tailwindcss.com/). Executing `blitz install tailwind` is all it takes to get up and running, and you can let Blitz do the rest.
```
Comment (flybayer) [inline docs/using-recipes.mdx:14]: The link for `blitz install` is wrong

File: docs/writing-recipes.mdx [hunk 5]
```diff
@@ -0,0 +1,126 @@
+---
+title: Writing Your Own Recipes
+sidebar_label: Writing Recipes
+---
+
+The incredible power of Blitz Recipes isn't limited to the recipes that the Blitz team has created. The API for building recipes is a public API (although one that is subject to change) exposed via the [`@blitzjs/installer`](https://npm.im/@blitzjs/installer) package, which can be installed into your own scripts to write a completely custom recipe. Combined with the power of [`recast`](https://github.com/benjamn/recast) for transforming existing files, fully automated code migrations are first-class citizens in the Blitz ecosystem. In fact, in the future we hope that upgrading Blitz will actually be possible with a quick `blitz install blitz@1.0.0`.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:6]:
```suggestion
The incredible power of Blitz Recipes isn't limited to the official recipes in the Blitz repo. The API for building recipes is a public API (although one that is subject to change) exposed via the [`@blitzjs/installer`](https://npm.im/@blitzjs/installer) package, which can be installed into your own scripts to write a completely custom recipe. Combined with the power of [`recast`](https://github.com/benjamn/recast) for transforming existing files, fully automated code migrations are first-class citizens in the Blitz ecosystem. In fact, in the future we hope that upgrading Blitz will actually be possible with a quick `blitz install blitz@1.0.0`.
```

File: docs/writing-recipes.mdx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+### Setup
+
+To author your own recipe, you'll want to create a new package and install a couple of dependencies. You'll only need the `ast-types` dependency if you're using a **transform** step to modify an existing file. If you're only creating new files or adding dependencies you'll only need `@blitzjs/installer`.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:10]: Would it be good to suggest `tsdx` here?

File: docs/writing-recipes.mdx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+```bash
+yarn add -EL @blitzjs/installer ast-types
+```
+
+The Recipe API all revolves around the `RecipeBuilder` factory. Blitz assumes that the file referenced in the `main` field of your `package.json` has a recipe as its default export, so we can go ahead and set that up.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:16]: Can you add a code snippet here of `package.json` showing the `main` field?

File: docs/writing-recipes.mdx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  stepName: string
+  explanation: string
+}
+```
+
+Eventually we expect to provide hooks into the recipe lifecycle, making some of the metadata like `stepId` critical.
+
+### Adding Dependencies
+
+The first action we can take is adding dependencies to the user's application. This step type will automatically detect whether the user is using `yarn` or `npm` and use the proper tool. The configuration is very straightforward - it accepts a list of packages, their versions, and whether or not they should be installed as a `devDependency`.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:56]: Em dash :) 
```suggestion
The first action we can take is adding dependencies to the user's application. This step type will automatically detect whether the user is using `yarn` or `npm` and use the proper tool. The configuration is very straightforward — it accepts a list of packages, their versions, and whether or not they should be installed as a `devDependency`.
```

File: docs/writing-recipes.mdx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+One incredibly powerful part of recipes is the ability to generate files from templates. By supplying the `templateValues` configuration, you can either supply a hard-coded object for values to interpolate in the template or a function that returns an object. The function will be passed any additional arguments passed to `blitz install` (e.g. `blitz install myrecipe --someConfig=false`). You can read more about our custom templating language [here.](templates) The template files can go anywhere in your recipe's file structure, you supply the path as a part of the recipe definition.
+
+```typescript
+builder.addNewFilesStep({
+  stepId: "addStyles",
+  stepName: "Add base Tailwind CSS styles",
+  explanation: `Next, we need to actually create some stylesheets! These stylesheets can either be modified to include global styles for your app, or you can stick to using classnames in your components.`,
+  targetDirectory: "./app",
+  templatePath: join(__dirname, "templates", "styles"),
+  templateValues: {},
```
Comment (flybayer) [inline docs/writing-recipes.mdx:81]: Where does `join` come from? We should include the import for that in this snippet.

File: docs/writing-recipes.mdx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  packages: [
+    {name: "tailwindcss", version: "1"},
+    {name: "postcss-preset-env", version: "latest", isDevDep: true},
+  ],
+})
+```
+
+### Adding files
+
+One incredibly powerful part of recipes is the ability to generate files from templates. By supplying the `templateValues` configuration, you can either supply a hard-coded object for values to interpolate in the template or a function that returns an object. The function will be passed any additional arguments passed to `blitz install` (e.g. `blitz install myrecipe --someConfig=false`). You can read more about our custom templating language [here.](templates) The template files can go anywhere in your recipe's file structure, you supply the path as a part of the recipe definition.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:72]:
```suggestion
One incredibly powerful part of recipes is the ability to generate files from templates. 

We use a custom templating language that's easy to both read and write. [Read our template documentation to learn how to write templates](templates).

By supplying the `templateValues` configuration, you can either supply a hard-coded object for values to interpolate in the template or a function that returns an object. The function will be passed any additional arguments passed to `blitz install` (e.g. `blitz install myrecipe --someConfig=false`). The template files can go anywhere in your recipe's file structure, you supply the path as a part of the recipe definition.
```

File: docs/writing-recipes.mdx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+  explanation: `Next, we need to actually create some stylesheets! These stylesheets can either be modified to include global styles for your app, or you can stick to using classnames in your components.`,
+  targetDirectory: "./app",
+  templatePath: join(__dirname, "templates", "styles"),
+  templateValues: {},
+})
+```
+
+### Modifying existing files
+
+Arguably the most powerful part of Blitz recipes, using Recast you can write a transform that will modify an existing file. The transform function is passed the AST representing the selected file, an object for building new nodes, and an object to assist with typechecking and assertions on nodes. The Recast documentation documents all of those objects well.
```
Comment (flybayer) [inline docs/writing-recipes.mdx:87]:
```suggestion
Arguably the most powerful part of Blitz recipes, using Recast you can write a transform that will modify an existing file. The transform function is passed the AST representing the selected file, an object for building new nodes, and an object to assist with typechecking and assertions on nodes. [The Recast documentation](https://github.com/benjamn/recast) documents all of those objects well.
```

File: docs/writing-recipes.mdx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+  stepName: "Import stylesheets",
+  explanation: `Finaly, we can import the stylesheets we created, into our application. For now we'll put them in document.tsx, but if you'd like to only style a part of your app with tailwind you could import the styles lower down in your component tree.`,
+  singleFileSearch: paths.app(),
+  transform(ast: ASTNode, b: builders, t: NamedTypes) {
+    const stylesImport = b.importDeclaration([], b.literal("app/styles/index.css"))
+    if (t.File.check(ast)) {
+      return addImport(ast, b, t, stylesImport)!
+    }
+    throw new Error("Not given valid source file")
+  },
```
Comment (flybayer) [inline docs/writing-recipes.mdx:105]: Where does `paths.app()` come from? Need to document that

File: docs/writing-recipes.mdx [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+})`
+expect(print(addImport(parse(sampleFile), b, t, newImport)).code).toMatchSnapshot()
+```
+
+### Publishing
+
+That's all you need to build a recipe! At this point, you can commit and push up to GitHub, and your recipe is available to the world. Users can install your recipe by passing your full repository name to `blitz install` - for example:
+
+```bash
+blitz install someuser/my-awesome-recipe
```
Comment (flybayer) [inline docs/writing-recipes.mdx:125]:
```suggestion
blitz install some-githubuser/my-awesome-recipe
```

File: docs/templates.mdx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+// Output (includeAnalytics: false)
+export default function __ModelName__Home() {
+  // ...
+}
+```
+
+---
+
+_Interested in helping with the templating language? There's plenty of room for improvement including supporting loops and iteration, actual evaluation of conditionals rather than only truthiness checks, and more! Join us in [Slack](https://slack.blitzjs.com)._
```
Comment (flybayer) [inline docs/templates.mdx:63]: 💯💯

File: docs/using-recipes.mdx (hunk 3 above)
Comment (aem) [reply docs/using-recipes.mdx:10]: That's fine, I liked that I could get the colorized output in there and wasn't sure if people would know what they were looking at if it wasn't a screenshot, but I definitely agree with the accessibility problems.

File: docs/using-recipes.mdx (hunk 4 above)
Comment (aem) [reply docs/using-recipes.mdx:14]: Oops yeah that was a placeholder because I meant to write up a page for `blitz install` too haha
